torch/ao/quantization/fx/convert.py

```diff
@@ -450,11 +450,12 @@ def _restore_state(
 ) -> Tuple[Dict[str, Tuple[str, type]],
            PrepareCustomConfig,
            Set[str]]:
-    assert _is_observed_module(observed), \
-        'incoming model must be produced by prepare_fx'
-    prepare_custom_config: PrepareCustomConfig = observed._prepare_custom_config  # type: ignore[assignment]
-    node_name_to_scope: Dict[str, Tuple[str, type]] = observed._node_name_to_scope  # type: ignore[assignment]
-    observed_node_names: Set[str] = observed._observed_node_names  # type: ignore[assignment]
+    # assert _is_observed_module(observed), \
```

Comment:
Can we have a dataclass
```python
@dataclass
class QuantizationConfig: # or other name
  prepare_custom_config: ...
  node_to_scope: ...
  observed_node_names:...
  ...
```
basically all those individual fields that need to be present for an Observed/Fused/Quantized module.

Then have one instance set into graph_module.meta['_quantized_config']

The pros is that 1. it's all in the same place and we know presence of it by checking one key in the meta dict.
and 2. code change is easier: for example: add observed = observed.meta before L454 then dont need to change anything after it as we kept the dot syntax with the dataclass.

Reply:
sure, sounds good

Reply:
actually, we will also have some preserved attributes as well requested by users, we still need to "pollute" the keys of model.meta..

Reply:
well, you can have a user_preserved_attr dict in the QuantizationConfig dataclass and use that instead?